contents/performance/WebPerformance.md

```diff
@@ -0,0 +1,79 @@
+
+**Optimize Images**
+
+Images form a [significant portion](https://httparchive.org/reports/page-weight) of the resources loaded on web applications. To improve performance and speed up the site for users, web developers should thus try to reduce the amount of bandwidth used to load images.
+
+Images can be optimized for the web in the following ways:
+- Use the [correct file format](https://medium.com/beginners-guide-to-mobile-web-development/web-image-formats-googles-webp-17e2fe5fc53e) for your images.
+- [Compress your images](https://www.html5rocks.com/en/tutorials/speed/img-compression/) to reduce the amount of data that needs to be loaded. Be careful to strike a balance between image size and quality for optimal user experience.
+- Deliver [appropriately sized images](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images) based on the resolution of the client device. This prevents smaller (often mobile or lower-end) devices from wasting bandwidth loading large images that cannot be displayed at their full resolution on the device.
+
+**Progressive Loading**
+
+The sooner the user sees content being displayed on the page, the faster they perceives the site to be. One way to reduce the time taken for content to be shown is to include only the critical parts of the page in the initial render, and progressively load the remaining resources and update the page as they become available. Though the page may not contain all the content upon the initial render, the partial render makes the site feel faster and more responsive to the user.
+
+Here are several ways how progressive loading can be effectively applied:
+- Load only the [most critical content](https://developers.google.com/web/fundamentals/performance/critical-rendering-path/) as part of the initial render, and fetch remaining resources asynchronously after the render. [Code splitting](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/code-splitting/) can also be helpful here to reduce the amount of Javascript that needs to be loaded for the initial render.
+- Use [lazy loading](https://developers.google.com/web/fundamentals/performance/lazy-loading-guidance/images-and-video/) to defer requesting non-critical resources until they are needed.
+- Show [content placeholders](https://medium.com/@praveencnaik/content-placeholder-the-new-design-trend-for-audience-involvement-e2ab533d7304) while fetching resources to indicate loading progress to the user.
+
+## Measuring Web Performance
+
+Improving web performance is a continuous, ongoing process. The ability to measure and track performance is necessary for developers to monitor the impact of their changes over time.
+
+Here are some ways that you can measure the performance of your web applications:
```

Comment:
Maybe you can consider including https://www.webpagetest.org/ here. They allow us to run a free website speed test from multiple locations around the globe using real browsers at real consumer connection speeds. So you can simulate how the web looks like to say a user using low-end phones with low internet bandwidth.

Reply:
Thanks for the suggestion! Looks like a really useful tool 🙂